metadata_backend/api/handlers/submission.py

```diff
@@ -541,15 +560,19 @@ async def publish_submission(self, req: Request) -> Response:
                 },
             }
             datasets_patch.append(patch_ds)
+        extra_info_patch.extend(datasets_patch)
+        extra_info_submission = await operator.update_submission(submission_id, extra_info_patch)
+
+        # we next try to publish the to Metax before actually publishing the submission
+        await metax_handler.update_dataset_with_doi_info(
+            await operator.read_submission(extra_info_submission), metax_ids
+        )
+        await metax_handler.publish_dataset(metax_ids)
 
-        # Create draft DOI ??? and delete draft objects from the submission
+        # Delete draft objects from the submission
         for obj in submission["drafts"]:
             await obj_ops.delete_metadata_object(obj["schema"], obj["accessionId"])
 
-        # update study to metax with data comming from doi info
-        await metax_handler.update_dataset_with_doi_info(submission["doiInfo"], metax_ids)
```

Comment:
Why did you remove this line?

Reply:
It was raised before the publication procedure starts to line 567, so nothing will be deleted before we know that metax is ok. Also had to extra DB call because data needed for relation mapping is added to `extra_info` -field